packages/preload/src/index.ts

```diff
@@ -923,6 +927,13 @@ function initExposure(): void {
     return ipcInvoke('menu-registry:getContributedMenus', context);
   });
 
+  contextBridge.exposeInMainWorld(
+    'getKubeGeneratorsIdsByType',
+    async (type: KubernetesGeneratorType): Promise<string[]> => {
+      return ipcInvoke('kube-generator-registry:getKubeGeneratorsIdsByType', type);
+    },
+  );
```

Comment:
we need a more rich structure on the client side, not only the id but also the supported type.

So for example we know if we have multiple generators supporting Pods or none, etc.

Reply:
Does replacing `KubernetesGeneratorType` with something like `KubernetesGenerators` wthout arguments, and simply returning an array of KubeGeneratorsInfo (name, id, types) is better ?

It would let the client filtering on their need.

Reply:
so expose `getKubeGenerators()` returning `KubeGeneratorsInfo[]` object having `name, id and types` property.

but it's fine to keep types[] as a parameter like client could request only Generators matching a given subset of types (optional arguments)
But that's fine to not provide the argument as well